lib/IRGen/IRGenSIL.cpp

```diff
@@ -5456,6 +5466,12 @@ void IRGenSILFunction::visitCondFailInst(swift::CondFailInst *i) {
   llvm::BasicBlock *contBB = llvm::BasicBlock::Create(IGM.getLLVMContext());
   Builder.CreateCondBr(cond, failBB, contBB);
   Builder.emitBlock(failBB);
+  // FIXME: This fails when used with -O
+  // TODO: If this bug is fixed, it may be better to put this code in emitTrap()
```

Comment:
Oh! Ok, I see that you have :). What's the failure with -O? Would it help to address that first?